src/Hangfire.PostgreSql/PostgreSqlJobQueue.cs

```diff
@@ -129,7 +129,9 @@ internal IFetchedJob Dequeue_Transaction(string[] queues, CancellationToken canc
         throw new ArgumentException("Queue array must be non-empty.", nameof(queues));
       }
 
-      long timeoutSeconds = (long)_storage.Options.InvisibilityTimeout.Negate().TotalSeconds;
+      long timeoutSeconds = _storage.Options.SlidingInvisibilityTimeout.HasValue
+        ? (long)_storage.Options.SlidingInvisibilityTimeout.Value.Negate().TotalSeconds
+        : (long)_storage.Options.InvisibilityTimeout.Negate().TotalSeconds;
```

Comment:
```suggestion
      long timeoutSeconds = (long)(_storage.Options.SlidingInvisibilityTimeout ?? _storage.Options.InvisibilityTimeout).Negate().TotalSeconds;
```
Something like that?